charts/trino/templates/ingress.yaml

```diff
@@ -0,0 +1,63 @@
+
+{{- if .Values.ingress.enabled -}}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ template "trino.coordinator" . }}
+  labels:
+    {{- include "trino.labels" . | nindent 4 }}
+  {{- with .Values.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if .Values.ingress.tls.secretName }}
+  tls:
+    - hosts:
+        - "{{ .Values.ingress.host }}"
+      secretName: {{ .Values.ingress.tls.secretName }}
+  {{- end }}
+  rules:
+  - host: "{{ .Values.ingress.host }}"
+    http:
+      paths:
+      - path: "/"
+        pathType: Prefix
+        backend:
+          service:
+            name: {{ include "trino.fullname" . }}
+            port:
+              number: {{ .Values.server.config.http.port }}
+{{- else -}}
+{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
+kind: Ingress
+metadata:
+  name: {{ template "trino.coordinator" . }}
+  labels:
+    {{- include "trino.labels" . | nindent 4 }}
+  {{- with .Values.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if .Values.ingress.tls.secretName }}
+  tls:
+    - hosts:
+        - "{{ .Values.ingress.host }}"
+      secretName: {{ .Values.ingress.tls.secretName }}
+  {{- end }}
+  rules:
+    - host: "{{ .Values.ingress.host }}"
+      http:
+        paths:
+          - path: "/*"
+            backend:
+              serviceName: {{ include "trino.fullname" . }}
+              servicePort: {{ .Values.server.config.http.port }}
```

Comment:
We use the http ports because ingress would terminate TLS?

If so then should we detect this and enable `http-server.process-forwarded` if ingress is enabled?

Reply:
ping @dragonslayer27